satellite/satellitedb/overlaycache.go

```diff
@@ -33,6 +33,80 @@ type overlaycache struct {
 	db *satelliteDB
 }
 
+// SelectAllStorageNodesUpload returns all nodes that qualify to store data, organized as reputable nodes and new nodes
+func (cache *overlaycache) SelectAllStorageNodesUpload(ctx context.Context, selectionCfg overlay.NodeSelectionConfig) (reputable, new []overlay.CachedNode, err error) {
+	defer mon.Task()(&ctx)(&err)
+	version, err := version.NewSemVer(selectionCfg.MinimumVersion)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	query := `
+		SELECT id, address, last_net, last_ip_port, true as reputable
```

Comment:
You can do a single query without union all with `SELECT id, address, last_net, last_ip_port, (total_audit_count < $4 OR total_uptime_count < $5) as isnew`